controllers/consoleplugin/consoleplugin_objects.go

```diff
@@ -347,9 +356,77 @@ func (b *builder) setLokiConfig(lconf *cfg.LokiConfig) {
 			}
 		}
 	}
-	if b.loki.UseHostToken() {
+	if lk.UseHostToken() {
 		lconf.TokenPath = b.volumes.AddToken(constants.PluginName)
 	}
+	return lconf
+}
+
+func (b *builder) getPromConfig(ctx context.Context) cfg.PrometheusConfig {
+	if !helper.UsePrometheus(b.desired) {
+		return cfg.PrometheusConfig{}
+	}
+
+	// Default config = manual
+	tls := b.desired.Prometheus.Querier.Manual.TLS
+	config := cfg.PrometheusConfig{
+		URL:              b.desired.Prometheus.Querier.Manual.URL,
+		ForwardUserToken: b.desired.Prometheus.Querier.Manual.ForwardUserToken,
+	}
+	if b.desired.Prometheus.Querier.Mode == flowslatest.PromModeAuto {
+		if b.info.UseOpenShiftSCC /* aka IsOpenShift */ {
+			config = cfg.PrometheusConfig{
+				URL:              "https://thanos-querier.openshift-monitoring.svc:9091/",
```

Comment:
r u sure the svc will always be at `9091` any better way than hardcoding it ?

Reply:
We could query for the service object, but we cannot avoid some hardcoded stuff (if not the port number, it will be the port name in YAML, and the service name itself). So, I'm not sure it's worth it. And anyway that's just the "Auto" mode: if the hardcoded config doesn't work anymore later, users can switch to manual config.

Reply:
I guess we can start like that and change when we see issues ports tends to get reused by different app sometimes can also change on different arch like we have seen with metrics on ppc64le, since u have a log stating we can't do auto mode and fall back to manual that will be good enough for now